chopper/lib/src/utils.dart

```diff
@@ -66,98 +66,19 @@ String mapToQuery(
   bool useBrackets = false,
   bool includeNullQueryVars = false,
 }) =>
-    _mapToQuery(
+    QS.encode(
       map,
-      useBrackets: useBrackets,
-      includeNullQueryVars: includeNullQueryVars,
-    ).join('&');
-
-Iterable<_Pair<String, String>> _mapToQuery(
-  Map<String, dynamic> map, {
-  String? prefix,
-  bool useBrackets = false,
-  bool includeNullQueryVars = false,
-}) {
-  final Set<_Pair<String, String>> pairs = {};
-
-  map.forEach((key, value) {
-    String name = Uri.encodeQueryComponent(key);
-
-    if (prefix != null) {
-      name = useBrackets
-          ? '$prefix${Uri.encodeQueryComponent('[')}$name${Uri.encodeQueryComponent(']')}'
-          : '$prefix.$name';
-    }
-
-    if (value != null) {
-      if (value is Iterable) {
-        pairs.addAll(_iterableToQuery(name, value, useBrackets: useBrackets));
-      } else if (value is Map<String, dynamic>) {
-        pairs.addAll(
-          _mapToQuery(
-            value,
-            prefix: name,
-            useBrackets: useBrackets,
-            includeNullQueryVars: includeNullQueryVars,
-          ),
-        );
-      } else {
-        pairs.add(
-          _Pair<String, String>(name, _normalizeValue(value)),
-        );
-      }
-    } else {
-      if (includeNullQueryVars) {
-        pairs.add(_Pair<String, String>(name, ''));
-      }
-    }
-  });
-
-  return pairs;
-}
-
-Iterable<_Pair<String, String>> _iterableToQuery(
-  String name,
-  Iterable values, {
-  bool useBrackets = false,
-}) =>
-    values.where((value) => value?.toString().isNotEmpty ?? false).map(
-          (value) => _Pair(
-            name,
-            _normalizeValue(value),
-            useBrackets: useBrackets,
-          ),
-        );
-
-String _normalizeValue(value) => Uri.encodeComponent(
-      value is DateTime
-          ? value.toUtc().toIso8601String()
-          : value?.toString() ?? '',
+      EncodeOptions(
+        listFormat: useBrackets ? ListFormat.brackets : ListFormat.repeat,
```

Comment:
If you want to deprecate `useBrackets`, consider adding the following:

```dart
String mapToQuery(
  Map<String, dynamic> map, {
  @Deprecated("Use listFormat instead. Will be removed in a future release.")
  bool useBrackets = false,
  bool includeNullQueryVars = false,
  ListFormat listFormat = ListFormat.repeat,
}) =>
    QS.encode(
      map,
      EncodeOptions(
        listFormat: useBrackets ? ListFormat.brackets : listFormat,
        allowDots: !useBrackets,
        encodeDotInKeys: !useBrackets,
        encodeValuesOnly: !useBrackets,
        skipNulls: !includeNullQueryVars,
        strictNullHandling: false,
        serializeDate: (DateTime date) => date.toUtc().toIso8601String(),
      ),
);
```

Reply:
Way ahead of you buddy :)